bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/AndroidDebugBridgeDevice.java

```diff
@@ -374,6 +380,225 @@ public void powerOffDevice()
         }
     }
 
+    public void startIntent(String command) throws NumberFormatException, AndroidDebugBridgeDeviceException,
+            ExecutionException, InterruptedException, TimeoutException {
+        String[] commandParts = command.split("\\|\\|");
+        String targetPackage = commandParts[0];
+        var targetPackageParts = targetPackage.split("/");
+        var packageName = targetPackageParts[0];
+        var activityName = targetPackageParts[1];
+        if (targetPackageParts.length == 1) {
+            if (!PACKAGE_NAME_PATTERN.matcher(packageName).matches()) {
+                logger.warn("{} is not a valid package name", packageName);
+                return;
+            }
+        } else if (targetPackageParts.length == 2) {
+            if (!PACKAGE_NAME_PATTERN.matcher(packageName).matches()) {
+                logger.warn("{} is not a valid package name", packageName);
+                return;
+            }
+            if (!SECURE_SHELL_INPUT_PATTERN.matcher(activityName).matches()) {
+                logger.warn("{} is not a valid activity name", activityName);
+                return;
+            }
+        } else {
+            throw new AndroidDebugBridgeDeviceException("Invalid target package " + targetPackage);
+        }
```

Comment:
Maybe this could be slightly rewritten for simplification and to avoid redundancy. Something like:

```
if (targetPackageParts.length > 2) {
    throw new AndroidDebugBridgeDeviceException("Invalid target package " + targetPackage);
}
if (!PACKAGE_NAME_PATTERN.matcher(packageName).matches()) {
    logger.warn("{} is not a valid package name", packageName);
    return;
}
if (targetPackageParts.length == 2 && !SECURE_SHELL_INPUT_PATTERN.matcher(activityName).matches()) {
    logger.warn("{} is not a valid activity name", activityName);
    return;
}
```